include/s3select_json_parser.h

```diff
@@ -563,8 +577,10 @@ class JsonParserHandler : public rapidjson::BaseReaderHandler<rapidjson::UTF8<>,
 	json_key_value_t key_value(key_path,v);
 	m_star_operation_cb(key_value);
       }
-	if (prefix_match)
-      		variable_match_operations.new_value(v);
+	if (prefix_match) {
+          json_key_value_t key_path_value(key_path,v);
+      		variable_match_operations.new_value(v, key_path_value); 
```

Comment:
a general note.
this method is quite intensive, each value related to SQL statement is passing through this method,
adding `key_path`  and `key_path_value`  increase the CPU usage.

for many use-cases the above is redundant (key_path is skipped)
one way to resolve this issue is to "push" the key-path only upon it is needed by the user.

a. user requests for JSON output format
b. the s3select-engine identifies what key-path is relevant to output-format according to SQL statement.

Reply:
it still missing the indication of whether the user requested  JSON-OUTPUT-FORAMT.
which means adding the key-path-value.